src/web/GestureHandler.ts

```diff
@@ -182,7 +182,12 @@ abstract class GestureHandler {
         // onHandlerStateChange only
         handlerTag: this.handlerTag,
         target: this.ref,
-        oldState: this.oldState,
+        // send oldState only when the state was changed, or is different than ACTIVE
+        // GestureDetector relies on the presence of `oldState` to differentiate between
+        // update events and state change events
+        ...(state !== this.previousState || state != 4
```

Comment:
can't we just make `oldState: undefined` if we don't want to include it? It'd look cleaner then destructing empty object imo

Reply:
Done in [c81b618](https://github.com/software-mansion/react-native-gesture-handler/pull/1934/commits/c81b6188d58362283437b00c594b6d085df1314a)